osgp/platform/osgp-adapter-ws-smartmetering/src/main/java/org/opensmartgridplatform/adapter/ws/smartmetering/endpoints/SmartMeteringMonitoringEndpoint.java

```diff
@@ -512,12 +516,70 @@ public ClearAlarmRegisterResponse getClearAlarmRegisterResponse(
         } catch (final FunctionalException e) {
             throw e;
         } catch (final Exception e) {
-            LOGGER.error("Exception: {} while sending clear alarm register request of device: {} for organisation {}.",
+            log.error("Exception: {} while sending clear alarm register request of device: {} for organisation {}.",
                     e.getMessage(), request.getDeviceIdentification(), organisationIdentification);
 
             this.handleException(e);
         }
         return response;
     }
+    
+    @PayloadRoot(localPart = "ActualPowerQualityRequest", namespace = SMARTMETER_MONITORING_NAMESPACE)
+    @ResponsePayload
+    public ActualPowerQualityAsyncResponse getActualPowerQuality(
+            @OrganisationIdentification final String organisationIdentification,
+            @RequestPayload final org.opensmartgridplatform.adapter.ws.schema.smartmetering.monitoring.ActualPowerQualityRequest request, @MessagePriority final String messagePriority,
+            @ResponseUrl final String responseUrl, @ScheduleTime final String scheduleTime) throws OsgpException {
+
+        log.debug("Incoming ActualPowerQualityRequest for meter: {}.", request.getDeviceIdentification());
+
+
+        ActualPowerQualityAsyncResponse response = null;
+
+        try {
+            final ActualPowerQualityRequest dataRequest = this.monitoringMapper
+                    .map(request,
+                            ActualPowerQualityRequest.class);
+
+            final String correlationUid = this.monitoringService.enqueueActualPowerQualityRequestData(
+                    organisationIdentification, request.getDeviceIdentification(), dataRequest, MessagePriorityEnum.getMessagePriority(messagePriority),
+                    this.monitoringMapper.map(scheduleTime, Long.class));
+
+            response = new ActualPowerQualityAsyncResponse();
+            response.setCorrelationUid(correlationUid);
+            response.setDeviceIdentification(request.getDeviceIdentification());
+            this.responseUrlService.saveResponseUrlIfNeeded(correlationUid, responseUrl);
+        } catch (final Exception e) {
+            log.error("Exception: {} while requesting meter reads for device: {} for organisation {}.",
```

Comment:
"Requesting meter reads" in the error message refers to another endpoint, and may be misleading to someone analyzing the error logs.

Reply:
changed